3723/1/page_printer.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'sorted_pages'
+
+class PagePrinter
+  attr_reader :rows, :named_rows
+
+  def initialize(rows)
+    @rows = rows
+    @named_rows = []
```

Comment:
It's not necessary to have this instance variable. Set a local one inside `filter_by_name` and return its value from the method. It's even better if you don't mutate `@rows` so you can get its value any time. Check out the idea:
```ruby
def call(option = nil)
  rows_to_print =
    if option.is_a? Integer
      filter_by_number(option) # it's also a filter, isn't it?
    elsif option.is_a? String
      filter_by_name(option)
    else
      @rows
    end
  
  print_info(rows_to_print)
end
```

But what I've already said and what I'd like to change is multiple responsibilities of the class. I'm sure that something like that would be better, especially in a long run:
```ruby
def call(option = nil)
  filtered_rows = filter_rows_by option
  
  print_as_table filtered_rows
end

private

def filter_rows_by(option)
  if option.is_a? String
    NameFilter.new(@rows, option).call # Another class which does filtration by name
  elsif option.is_a? Integer
    NumberFilter.new(@rows, option).call # Another class which does filtration by number
  else
    @rows
  end
end
```